shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala

```diff
@@ -105,7 +106,9 @@ class UCX(transport: UCXShuffleTransport, executor: BlockManagerId, rapidsConf:
       new ThreadFactoryBuilder()
         .setNameFormat("progress-thread-%d")
         .setDaemon(true)
-        .build))
+        .build,
+      () => RmmSpark.associateCurrentThreadWithShuffle(),
```

Comment:
nit: will eta-expansion work here and other places?

```suggestion
      RmmSpark.associateCurrentThreadWithShuffle(_),
```

Reply:
No. There is nothing passed in so we cannot use `(_)`. I can remove it and make it just be
```
RmmSpark.associateCurrentThreadWithShuffle
```

But I don't like that because it is not obvious that it is a function being passed in.  It is just a bit too magic for me.